actors/cloudbroker/cloudbroker__computenode/methodclass/cloudbroker_computenode.py

```diff
@@ -183,38 +179,35 @@ def maintenance(self, id, gid, vmaction, **kwargs):
         return True
 
     def _stop_vfws(self, stack, title, ctx):
-        vfws = self._vcl.search({'gid': stack['gid'],
-                                 'nid': int(stack['referenceId'])})[1:]
-        for vfw in vfws:
-            ctx.events.sendMessage(title, 'Stopping Virtual Firewal %s' % vfw['id'])
-            self.cb.netmgr.fw_stop(vfw['guid'])
+        cloudspaces = self.models.Cloudspace.objects(stack=stack, status__in=["DEPLOYED", "VIRTUAL"])
+        for cloudspace in cloudspaces:
+            ctx.events.sendMessage(title, 'Stopping Virtual Firewal %s' % cloudspace.id)
+            self.cb.netmgr.destroy(cloudspace)
 
     def _start_vfws(self, stack, title, ctx):
-        vfws = self._vcl.search({'gid': stack['gid'],
-                                 'nid': int(stack['referenceId'])})[1:]
-        for vfw in vfws:
-            ctx.events.sendMessage(title, 'Starting Virtual Firewal %s' % vfw['id'])
-            self.cb.netmgr.fw_start(vfw['guid'])
+        cloudspaces = self.models.Cloudspace.objects(stack=stack, status='DEPLOYED')
```

Comment:
should also start when in status virtual and update the status to deployed when successfull